File: sdks/php/src/Resources/Analytics.php
<?php

declare(strict_types=1);

namespace HookSniff\Resources;

use HookSniff\Request;

/**
 * HookSniff API Resource: Analytics
 */
class Analytics
{
    /**
     * @param array{baseUrl: string, token: string, timeout: int, numRetries: int} $ctx
     */
    public function __construct(private readonly array $ctx) {}

    /**
     * Get delivery trend data.
     *
     * @return array{data: array<int, array{date: string, total: int, delivered: int, failed: int}>}
     * @throws \HookSniff\ApiException
     */
    public function trends(?string $since = null, ?string $until = null): array
    {
        $req = new Request('GET', '/v1/analytics/deliveries');
        $params = [];
        if ($since !== null) $params['since'] = $since;
        if ($until !== null) $params['until'] = $until;
        if (!empty($params)) $req->setQueryParams($params);
        return $req->send($this->ctx);
    }

    /**
     * Get success rate metrics.
     *
     * @return array{rate: float, total: int, delivered: int, failed: int}
     * @throws \HookSniff\ApiException
     */
    public function successRate(?string $since = null, ?string $until = null): array
    {
        $req = new Request('GET', '/v1/analytics/success-rate');
        $params = [];
        if ($since !== null) $params['since'] = $since;
        if ($until !== null) $params['until'] = $until;
        if (!empty($params)) $req->setQueryParams($params);
        return $req->send($this->ctx);
    }

    /**
     * Get latency metrics.
     *
     * @return array{p50: float, p95: float, p99: float, avg: float}
     * @throws \HookSniff\ApiException
     */
    public function latency(?string $since = null, ?string $until = null): array
    {
        $req = new Request('GET', '/v1/analytics/latency');
        $params = [];
        if ($since !== null) $params['since'] = $since;
        if ($until !== null) $params['until'] = $until;
        if (!empty($params)) $req->setQueryParams($params);
        return $req->send($this->ctx);
    }
}

File: sdks/java/src/main/java/com/hooksniff/resources/Analytics.java
package com.hooksniff.resources;

import com.hooksniff.ApiException;
import com.hooksniff.HookSniffConfig;
import com.hooksniff.RequestHelper;
import com.google.gson.reflect.TypeToken;

import java.io.IOException;
import java.lang.reflect.Type;
import java.util.Map;

/**
 * Delivery analytics and statistics.
 */
public class Analytics {
    private static final Type MAP = new TypeToken<Map<String, Object>>(){}.getType();

    private final HookSniffConfig config;

    public Analytics(HookSniffConfig config) {
        this.config = config;
    }

    /** Get delivery statistics */
    public Map<String, Object> stats() throws ApiException, IOException, InterruptedException {
        RequestHelper req = new RequestHelper("GET", "/v1/analytics/stats");
        return req.send(config, MAP);
    }

    /** Get delivery trends */
    public Map<String, Object> trends() throws ApiException, IOException, InterruptedException {
        RequestHelper req = new RequestHelper("GET", "/v1/analytics/trends");
        return req.send(config, MAP);
    }

    /** Get success rate */
    public Map<String, Object> successRate() throws ApiException, IOException, InterruptedException {
        RequestHelper req = new RequestHelper("GET", "/v1/analytics/success-rate");
        return req.send(config, MAP);
    }

    /** Get latency data */
    public Map<String, Object> latency() throws ApiException, IOException, InterruptedException {
        RequestHelper req = new RequestHelper("GET", "/v1/analytics/latency");
        return req.send(config, MAP);
    }
}
